### PR TITLE
ir_transmitter: add support for repeated commands in NEC protocol

### PR DESCRIPTION
**Pull request by ngilles**
# What does this implement/fix?

I have a fan controllable through IR that seemingly was using the NEC protocol, however it actually sends 2 instances of the `command` bits. This patch provides support for sending and recieving these types of transmissions.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3069

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [X] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
...
        # Send ON/OFF toggle (NEC: address=0xBF40, command=0xFA05 command_repeats=2)
        - remote_transmitter.transmit_nec:
            address: 0xBF40
            command: 0xFA05
            command_repeats: 2
...
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs): https://github.com/esphome/esphome-docs/pull/3069

### CODE REVIEW

**Comment by nielsnl68** [comment]
@ngilles could you please fix the above issues, i like to test it to see of this fixes an issue i have will fix with this PR.

**Comment by nagyrobi** [comment]
I think this could be useful for all protocols, not only NEC.

**Review by jesserockz — COMMENTED** [review]
All remote_transmitter actions already have `repeat` as an option...
https://esphome.io/components/remote_transmitter#remote-transmitter-actions

Edit: Oh, this would be a repeat only for the `command`... :thinking:

**Comment by ngilles** [comment]
@jesserockz yeah the common `repeat` is sending the full packet multiple times, while this patch is for handling a variant of the NEC protocol where the command bits are repeated within a single packet.

It was quite annoying cause most IR decoders were just saying "yep this is NEC alright with <address> and <command>", yet when trying to control the fan, I couldn't :-D

Finally I dug out my old cheapo logic analyzer and attached it to the fan's remote control IR LED and saw that extra few bits...

Going to look into the conflict when I get some time...

File: esphome/components/remote_base/__init__.py [hunk 1]
```diff
@@ -576,6 +577,7 @@ async def magiquest_action(var, config, args):
     {
         cv.Required(CONF_ADDRESS): cv.hex_uint16_t,
         cv.Required(CONF_COMMAND): cv.hex_uint16_t,
+        cv.Optional(CONF_COMMAND_REPEATS, default=1): cv.hex_uint16_t,
```
Comment (jesserockz) [inline esphome/components/remote_base/__init__.py]: No need for hex here
```suggestion
        cv.Optional(CONF_COMMAND_REPEATS, default=1): cv.uint16_t,
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**